docs/api/migration-guides/qiskit-1.0.mdx

```diff
@@ -0,0 +1,594 @@
+---
+title: Qiskit 1.0 Migration Guide
+description: Adapt to changes in Qiskit 1.0
+---
+
+# Qiskit 1.0 Migration Guide
+
+
+## Packaging changes
+
+Qiskit 1.0 uses a different packaging structure to previous Qiskits, and may cause problems in environments using packages that are not ready for Qiskit 1.0.
+
+<Admonition type="caution">
+We **do not** recommend attempting to upgrade an existing Python virtual environment from Qiskit 0.45 or before to Qiskit 1.0 in place.
+</Admonition>
+
+
+### For users
+
+You must start a new virtual environment to install Qiskit 1.0.
+It is very tricky and error-prone to upgrade an existing installation in-place to Qiskit 1.0.
+
+<Admonition type="info">
+This section will use the `venv` module that is part of the Python standard library in its examples.
+If you use a different tool (such as `virtualenv` or `conda`), you should consult its documentation if you are unsure of how the commands line up.
+
+The examples will assume a `bash`-like shell for non-Python command-line inputs.
+</Admonition>
+
+<span id="creating-a-venv"></span>
+#### Creating the new environment
+
+1. Create a new virtual environment, using your preferred (supported!) version of Python 3:
+
+   ```bash
+   python3 -m venv ~/qiskit-1.0-venv
+   ```
+
+   You can use any path you like in place of `~/qiskit-1.0-venv`.
+
+2. Activate the environment.
+
+   On Linux and macOS:
+   ```bash
+   source ~/qiskit-1.0-venv/bin/activate
+   ```
+
+   On Windows:
+   ```bash
+   source ~\qiskit-1.0-venv\Scripts\activate
+   ```
+
+3. Install all the packages you want to install.
+   You should do this using only one `pip install` command with all the dependencies on it.
+
+   ```bash
+   pip install 'qiskit>=1'
+   ```
+
+   You can include other packages you want installed alongside `qiskit` here by writing them as following arguments, such as
+
+   ```bash
+   pip install 'qiskit>=1' jupyterlab pandas matplotlib
+   ```
+
+4. If you are not planning to use the environment immediately, use the `deactivate` command to leave it.
+
```

Comment:
The only potential gotcha here for users that I think is good to call out here is compatibility with other dependencies that depend on qiskit-terra and/or `qiskit<1`. I think this will trip up users the most more than not being able to `pip install -U qiskit` because it's the piece that requires understanding how pip does dep solving or lack thereof and people doing things like `python -m venv test && test/bin/pip install qiskit>=1.0 && test/bin/pip install qiskit-qubit-reuse` (this is a good reminder I should update the plugin package to not depend on terra before 1.0). This will end up in a mixed installation state between qiskit 1.0.0 and qiskit-terra 0.46.0. I think this really should be called out in the top and not just in the troubleshooting guide below.

Reply:
I've thrown in an admonition warning about this, including a link to the troubleshooting section if people do get into problems in 75f6e3a.  Do you think it's enough?

I don't necessarily want to get too pessimistic up at the top of the guide - if they're just using IBM tooling (which I imagine most users are), they're _hopefully_ not going to have any issues by the time 1.0 is actually released.